robot/impl/resource_manager.go

```diff
@@ -472,6 +472,14 @@ func (manager *resourceManager) completeConfig(
 		if !ok || !gNode.NeedsReconfigure() {
 			continue
 		}
+		if !gNode.ShouldTryReconfigure() {
+			err := errors.Errorf("Reconfiguration error: reached max reconfiguration attempts for %s", gNode.Config().Name)
+			manager.logger.Error(err)
+			gNode.SetLastError(err)
+			gNode.MarkForRemoval()
```

Comment:
why set MarkForRemoval?

Reply:
I added this because without it, it would keep logging the reached max reconfig attempts error repeatedly. We can remove it if the aforementioned is fine or something else is affected by it

Reply:
Agree. I think we don't want the resource in the graph anymore if we've confirmed it can't reconfigure itself after 5 tries. If the user restarts their server or makes some change to the offending resource, the resource will be added again to the resource graph and it can re-attempt its configuration.

Reply:
but if we're using a cloud config, the broken module will just come back every ten seconds, right? since there'll be a diff if we removed it from the graph?